src/components/RoomDropdown/index.tsx

```diff
@@ -152,7 +158,7 @@ const RoomDropdown = ({ button }: Props) => {
     useEffect(() => {
         function handleClickOutside(event: any) {
             if (ref.current && !ref.current.contains(event.target)) {
-                setIsOpen(false)
+                toggleOpen()
```

Comment:
This change will make its `useEffect` run many times too often if we get the deps right.

The easiest way here?
```js
setIsOpen((current) => !current)
```
And this is how `toggleOpen` should look like in general. Otherwise you may end up inverting stale state.

I see that `setIsOpen` is not used anywhere else. Only in `toggleOpen`. We can make it better then with `useReducer`. You'll love it.

```js
const [isOpen, toggleOpen] = useReducer((current) => !current, false)
```

`toggleOpen` does not have to be listed in `useEffect` deps. That's good. And you call it w/o arguments. Just `toggleOpen`.

That's all.